pkg/processor/connection.go

```diff
@@ -0,0 +1,124 @@
+// Copyright 2025 Google LLC All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package processor
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+	"google.golang.org/grpc/health/grpc_health_v1"
+
+	allocationpb "agones.dev/agones/pkg/allocation/go"
+)
+
+// connectAndRun handles the full connection lifecycle to the processor service
+// It establishes a connection, creates a stream, registers the client, and then
+// delegates to handleStream to process messages until an error or cancellation
+func (p *processorClient) connectAndRun(ctx context.Context) error {
+	// Connect to the processor
+	conn, err := p.connect(ctx)
+	if err != nil {
+		return fmt.Errorf("failed to connect: %w", err)
```

Comment:
```suggestion
		return errors.Wrap(err, "failed to connect")
```

Use the errors library plz. Legacy libraries ftw 😄  (same for the others below)